soh/src/overlays/actors/ovl_En_Gm/z_en_gm.c

```diff
@@ -209,6 +209,10 @@ void func_80A3DC44(EnGm* this, GlobalContext* globalCtx) {
                 return;
             case 1:
                 gSaveContext.infTable[11] |= 2;
+                if (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SHUFFLE_MERCHANTS) &&
+                    !Flags_GetRandomizerInf(RAND_INF_MERCHANTS_MEDIGORON)) {
+                        gSaveContext.infTable[11] &= ~2;
+                    }
```

Comment:
What is the intent here? I'm lacking context sorry 😄

Reply:
This resets the "talked to Medigoron" flag in `infTable`, which has him do the initial sword conversation rather than the follow-up conversation.

Reply:
@dcvz @briaguya-ai Done.